brandon/server.js

```diff
@@ -0,0 +1,234 @@
+app.use(express.static('./public'));
+
+app.get('/', function(request, response) {
+  response.sendFile('index.html', { root: './public' });
+});
+
+app.get('/new', function(request, response) {
+  response.sendFile('new.html', { root: './public' });
+});
+
+app.get('/articles', function (request, response) {
+  // REVIEW: This query will join the data together from our tables and send it back to the client.
+  // TODO: Write a SQL query which joins all data from articles and authors tables on the author_id value of each
+  client.query(
+    `SELECT * FROM articles
+    INNER JOIN authors
+    ON authors.author_id = articles.author_id`)
+  .then(function(result) {
+    response.send(result.rows);
+  })
+  .catch(function (err) {
+    console.error(err)
+  });
+});
+
+app.post('/articles', function (request, response) {
+  // DONE: Write a SQL query to insert a new author, ON CONFLICT DO NOTHING
+  // DONE: Add author and "authorUrl" as data for the SQL query to interpolate.
+  //       Remember that client.query accepts two arguments: your SQL string and
+  //       an array of values that it will replace in a 1-to-1 relationship
+  //       with our placeholder values, signified with the syntax $1, $2, etc.
+  client.query(
+    `INSERT INTO authors (author, 'authorUrl')
+    VALUES($1, $2) ON CONFLICT DO NOTHING`,
+    [request.body.author, request.body.authorUrl]
+  )
+  .then(function() {
+    // DONE: Write a SQL query to insert a new article, using a sub-query to
+    // retrieve the author_id from the authors table. HINT: How might we combine
+    // the functionality of a SELECT with VALUES when inserting new rows?
+    // DONE: Add the required values from the request as data for the SQL query to interpolate
+    client.query(
+      `INSERT INTO articles(title, "authorUrl", category, "publishedOn", body)
+      SELECT author_id, $1, $2, $3, $4, $5
+      FROM authors
+      WHERE author = $6`,
+      [
+        request.body.title,
+        request.body.authorUrl,
+        request.body.category,
+        request.body.publishedOn,
+        request.body.body,
+        request.body.author,
+      ]
+    )
+  })
+  .then(function() {
+    response.send('Insert complete')
+  })
+  .catch(function(err) {
+    console.error(err)
+  });
+});
+
+app.put('/articles/:id', function(request, response) {
+  // DONE: Write a SQL query to update an author record. Remember that our articles now have
+  // an author_id property, so we can reference it from the request.body.
+  // DONE: Add the required values from the request as data for the SQL query to interpolate
+  client.query(
+    `UPDATE author
+    SET
+      title=$1, author=$2, "authorUrl"=$3, category=$4, "publishedOn"=$5, body=$6
+      WHERE author_id=$7
+      `,
+    [
+      request.body.title,
+      request.body.author,
+      request.body.authorUrl,
+      request.body.category,
+      request.body.publishedOn,
+      request.body.body,
+      request.params.id,
+    ]
+  )
```

Comment:
So on this part of the update you don't actually need to set all these values - look at what you're updating for the author (hint: you just need three values).